### DISCUSSION TITLE
使用绘制组件添加的图形，如何在点击时获取到绘制的数据或者图形对象

### DISCUSSION THREAD

**Discussion by webdq** [zouyaoji/vue-cesium#177] [Q&A] [answered]
使用绘制组件添加的图形，如何在点击时获取到绘制的数据或者图形对象，想添加弹出框修改属性，做双向绑定

**Comment by zouyaoji** [comment]
我看了一下现在版本可能不太好拿到啊。

最终目标是通过 ref 目标引用 获取到 renderDatas，

drawingsRef.getDrawingActionInstances()[3].cmpRef.value.renderDatas 去改。

只有配合 vc-selection-indicator 组件获取到绘制的图形对象，里面挂了索引 _vcPolylineIndex.

我下个版本加强一下哈，看看能不能给绘制的对象绑定些事件返回renderDatas

**Comment by zouyaoji** [comment, marked as answer]
下个版本 可以用  vc-selection-indicator 的 pickEvt 事件  或者 自己针对要编辑的绘制实例传事件进去。
![image](https://user-images.githubusercontent.com/24430116/161372861-83011f85-8904-4199-a506-cbdae807314f.png)

在事件中获取索引的方式
`e.pickedFeature.primitive._vcPolylineIndex || e.pickedFeature.primitive._vcParent._vcPolylineIndex` 

拿到索引后去对应的renderDatas里面去改就OK了

**Reply by webdq** [reply]
可以在每个标绘的 renderDatas 数据里返回 id 字段吗？

**Reply by zouyaoji** [reply]
使用场景是？

**Reply by webdq** [reply]
标绘的图形可能需要关联一些业务数据，或者能在绘制的图形上扩展额外数据/属性吗？

**Reply by zouyaoji** [reply]
估计可以，你尝试一下呢。
传一个 onReady回调函数。
![image](https://user-images.githubusercontent.com/24430116/164380908-a4427f11-dfef-4f6e-9d10-ae0bf452a169.png)
